components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/JsPolyglotGraphBuilder.java

```diff
@@ -118,176 +132,33 @@ public JsPolyglotGraphBuilder createWith(String script) {
                         .JS_FUNC_ON_LOGIN_REQUEST + " function is not defined.");
                 return this;
             }
-            onLoginRequestFn.apply(new Object[]{new GraalJsAuthenticationContext(authenticationContext)});
-            JsPolyglotGraphBuilderFactory.persistCurrentContext(authenticationContext, engine);
-        } catch (ScriptException | PolyglotException e) {
+            onLoginRequestFn.executeVoid(new GraalJsAuthenticationContext(authenticationContext));
+            JsPolyglotGraphBuilderFactory.persistCurrentContext(authenticationContext, context);
+        } catch (PolyglotException e) {
             result.setBuildSuccessful(false);
             result.setErrorReason("Error in executing the Javascript. " + FrameworkConstants.JSAttributes
                     .JS_FUNC_ON_LOGIN_REQUEST + " reason, " + e.getMessage());
             result.setError(e);
             if (log.isDebugEnabled()) {
                 log.debug("Error in executing the Javascript.", e);
             }
-        } catch (FrameworkException e) {
-            result.setBuildSuccessful(false);
-            result.setErrorReason("Error in restoring javascript context. " + FrameworkConstants.JSAttributes
-                    .JS_FUNC_ON_LOGIN_REQUEST + " reason, " + e.getMessage());
-            result.setError(e);
-            if (log.isDebugEnabled()) {
-                log.debug("Error in restoring the Javascript context.", e);
-            }
+        }
+
+        catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Let's add some info indicating the possibility of getting this error in log.error() and log `e`

Reply:
corrected